src/lib/device/device_manager.cpp

```diff
@@ -211,6 +211,14 @@ shared< VirtualDev > DeviceManager::create_vdev(vdev_parameters&& vparam) {
         LOGINFO("{} Virtual device is attempted to be created with size={}, it needs to be rounded to new_size={}",
                 vparam.vdev_name, in_bytes(input_vdev_size), in_bytes(vparam.vdev_size));
     }
+
+    // Adjust the maximum number chunks requested.
+    uint32_t max_num_chunks = 0;
+    for (const auto& d : m_dev_infos) {
+        max_num_chunks += hs_super_blk::max_chunks_in_pdev(d);
+    }
+
+    vparam.num_chunks = std::min(vparam.num_chunks, max_num_chunks);
```

Comment:
Can you please add a log here (if it is actually adjusted) saying number of chunks has been modified (like message in Line 211)